testing/scenario_app/ios/Scenarios/Scenarios/AppDelegate.m

```diff
@@ -19,31 +19,55 @@ - (BOOL)application:(UIApplication*)application
 
   // This argument is used by the XCUITest for Platform Views so that the app
   // under test will create platform views.
-  if ([[[NSProcessInfo processInfo] arguments] containsObject:@"--platform-view"]) {
-    FlutterEngine* engine = [[FlutterEngine alloc] initWithName:@"PlatformViewTest" project:nil];
-    [engine runWithEntrypoint:nil];
-
-    FlutterViewController* flutterViewController =
-        [[NoStatusBarFlutterViewController alloc] initWithEngine:engine nibName:nil bundle:nil];
-    [engine.binaryMessenger
-        setMessageHandlerOnChannel:@"scenario_status"
-              binaryMessageHandler:^(NSData* _Nullable message, FlutterBinaryReply _Nonnull reply) {
-                [engine.binaryMessenger
-                    sendOnChannel:@"set_scenario"
-                          message:[@"text_platform_view" dataUsingEncoding:NSUTF8StringEncoding]];
-              }];
-    TextPlatformViewFactory* textPlatformViewFactory =
-        [[TextPlatformViewFactory alloc] initWithMessenger:flutterViewController.binaryMessenger];
-    NSObject<FlutterPluginRegistrar>* registrar =
-        [flutterViewController.engine registrarForPlugin:@"scenarios/TextPlatformViewPlugin"];
-    [registrar registerViewFactory:textPlatformViewFactory withId:@"scenarios/textPlatformView"];
-    self.window.rootViewController = flutterViewController;
-  } else {
+  // The launchArgsMap should match the one in the `PlatformVieGoldenTestManager`.
+  static NSDictionary* launchArgsMap;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    launchArgsMap = @{
+      @"--platform-view" : @"platform_view",
+      @"--platform-view-cliprect" : @"platform_view_cliprect",
+      @"--platform-view-cliprrect" : @"platform_view_cliprrect",
+      @"--platform-view-clippath" : @"platform_view_clippath",
+      @"--platform-view-transform" : @"platform_view_transform",
+      @"--platform-view-opacity" : @"platform_view_opacity",
+    };
+  });
+  BOOL hasGoldenLaunchArg = NO;
+  for (NSString* key in launchArgsMap.allKeys) {
```

Comment:
Use fast-enumeration for iterating dictionaries instead of allocating a new array and then doing a value lookup later.
```
__block BOOL hasGoldenLaunchArg = NO;
[launchArgsMap enumerateKeysAndObjectsUsingBlock:^(NSString* argument, NSString* testName, BOOL *stop) {
   if ([[[NSProcessInfo processInfo] arguments] containsObject:argument]) {
     [self readyContextForPlatformViewTests:testName];
     hasGoldenLaunchArg = YES;
     *stop = YES;
  }
}];
```